drivers/eks/eks_driver.go

```diff
@@ -958,15 +957,30 @@ func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, options *t
 		return nil, err
 	}
 
+	var sendUpdate bool
 	if newState.KubernetesVersion != "" &&
 		newState.KubernetesVersion != state.KubernetesVersion {
+		sendUpdate = true
 		state.KubernetesVersion = newState.KubernetesVersion
 	}
 
-	if !reflect.DeepEqual(state, *oldstate) {
+	if newState.ClientID != "" && newState.ClientID != state.ClientID {
+		state.ClientID = newState.ClientID
+	}
+
+	if newState.ClientSecret != "" && newState.ClientSecret != state.ClientSecret {
+		state.ClientSecret = newState.ClientSecret
+	}
+
+	if sendUpdate {
```

Comment:
Do you know if there was a reason we previously sent an update if deepEqual wasn't true ? Rather than just checking version, since all we are really doing is updating the version. For id/secret seems that you don't need to because you are just switching which existing cloud key you want to use from rancher.

Reply:
I'm guessing that was done in anticipation of more things being configurable in the future.

Reply:
also, the desired state was just a copy of the original state with a desired version set, so it only would send if version was differed but was less efficient